planner/core/preprocess.go

```diff
@@ -132,9 +126,13 @@ const (
 
 // PreprocessorReturn is used to retain information obtained in the preprocessor.
 type PreprocessorReturn struct {
-	SnapshotTS        uint64
-	ExplicitStaleness bool
-	InfoSchema        infoschema.InfoSchema
+	initedLastSnapshotTS bool
+	ExplicitStaleness    bool
+	SnapshotTS           func(sessionctx.Context) (uint64, error)
```

Comment:
How about renaming it？ maybe `SnapshotTSFunc` or others

Reply:
Renamed to `SnapshotTSEvaluator`.